shotover-proxy/src/frame/cassandra.rs

```diff
@@ -333,21 +356,27 @@ pub enum CassandraOperation {
 
 impl CassandraOperation {
     /// Return all queries contained within CassandaOperation::Query and CassandraOperation::Batch
-    /// An Err is returned if the operation cannot contain queries or the queries failed to parse.
     ///
     /// TODO: This will return a custom iterator type when BATCH support is added
-    pub fn queries(&mut self) -> Result<std::iter::Once<&mut Statement>> {
-        match self {
-            CassandraOperation::Query {
-                query: CQL::Parsed(query),
-                ..
-            } => Ok(std::iter::once(query)),
-            CassandraOperation::Query {
-                query: CQL::FailedToParse(_),
-                ..
-            } => Err(anyhow!("Couldnt parse query")),
-            // TODO: Return CassandraOperation::Batch queries once we add BATCH parsing to cassandra-protocol
-            _ => Err(anyhow!("This operation cannot contain queries")),
+    pub fn queries(&mut self) -> Vec<&mut CassandraStatement> {
+        if let CassandraOperation::Query { query: cql, .. } = self {
+            cql.statements
+                .iter_mut()
+                .map(|stmt| &mut stmt.statement)
+                .collect()
+        } else {
+            Vec::<&mut CassandraStatement>::new()
+        }
+    }
+
+    /// Return all queries contained within CassandaOperation::Query and CassandraOperation::Batch
+    ///
+    /// TODO: This will return a custom iterator type when BATCH support is added
+    pub fn get_cql_statements(&mut self) -> Vec<&mut CQLStatement> {
```

Comment:
Now that we have removed has_error I was able to remove this by doing:
![image](https://user-images.githubusercontent.com/5120858/167750831-688950ea-8997-4cb7-ab59-69faca4d7e23.png)